desc/objectives/objective_funs.py

```diff
@@ -436,12 +453,23 @@ def compute_unscaled(self, x, constants=None):
         if constants is None:
             constants = self.constants
         assert len(params) == len(constants) == len(self.objectives)
-        f = jnp.concatenate(
-            [
-                obj.compute_unscaled(*par, constants=const)
-                for par, obj, const in zip(params, self.objectives, constants)
-            ]
-        )
+        if not self._is_multi_device:
+            f = jnp.concatenate(
+                [
+                    obj.compute_unscaled(*par, constants=const)
+                    for par, obj, const in zip(params, self.objectives, constants)
+                ]
+            )
+        else:
+            f = pconcat(
+                [
+                    obj.compute_unscaled(
```

Comment:
see if we can change this around to make JAX do it asynchronously instead of serially

Reply:
maybe removing this from list comprehension will help?